lib/ukdebug/print.c

```diff
@@ -93,28 +96,47 @@
 /* special level for printk redirection, used internally only */
 #define KLVL_DEBUG (-1)
 
-typedef int (*_ukplat_cout_t)(const char *, unsigned int);
+typedef __ssz (*cout_func)(const char *buf, __sz len);
 
 struct _vprint_console {
-	_ukplat_cout_t cout;
+	cout_func cout;
 	int newline;
 	int prevlvl;
 };
 
 /* Console state for kernel output */
 #if CONFIG_LIBUKDEBUG_REDIR_PRINTD || CONFIG_LIBUKDEBUG_PRINTK
-static struct _vprint_console kern  = { .cout = ukplat_coutk,
-					.newline = 1,
-					.prevlvl = INT_MIN };
+static struct _vprint_console kern  = {
+#if CONFIG_LIBUKCONSOLE
+	.cout = uk_console_out,
+#else
+	.cout = NULL,
+#endif /* CONFIG_LIBUKCONSOLE */
+	.newline = 1,
+	.prevlvl = INT_MIN
+};
 #endif
 
 /* Console state for debug output */
 #if !CONFIG_LIBUKDEBUG_REDIR_PRINTD
-static struct _vprint_console debug = { .cout = ukplat_coutd,
-					.newline = 1,
-					.prevlvl = INT_MIN };
+static struct _vprint_console debug = {
+#if CONFIG_LIBUKCONSOLE
```

Comment:
IMO this looks a bit ugly, it would be cleaner to:
```c
#if CONFIG_LIBUKCONSOLE
#define UK_COUT_FUNC        uk_console_out
#else /* CONFIG_LIBUKCONSOLE */
#define UK_COUT_FUNC        NULL
#endif /* CONFIG_LIBUKCONSOLE */
```

and then define your vprint stuctures as:
```c
static struct _vprint_console kern  = {
	.cout = UK_COUT_FUNC,
	.newline = 1,
	.prevlvl = INT_MIN
};

static struct _vprint_console debug = {
	.cout = UK_COUT_FUNC,
	.newline = 1,
	.prevlvl = INT_MIN
};
```

Reply:
I opted for the current option because it localized the change to the struct initialization and makes it more apparent what's going on.

Reply:
Can we resolve this?

Reply:
Yes, it's okay to resolve 👍🏼